src/components/Tools/ProgressTracker.vue

```diff
@@ -0,0 +1,269 @@
+        let totalRequired = 0;
+        let totalCompleted = 0;
+        r.reqs.forEach(req => {
+          if (!(req.fulfillment.fulfilledBy === 'self-check')) {
+            totalRequired += 1 + Object.values(req.fulfillment.additionalRequirements ?? {}).length;
+          }
+          [req.fulfillment, ...Object.values(req.fulfillment.additionalRequirements ?? {})].forEach(
+            reqOrNestedReq => {
+              if (reqOrNestedReq.safeMinCountFulfilled >= reqOrNestedReq.minCountRequired) {
+                totalCompleted += 1;
+              } else {
+                totalCompleted +=
+                  reqOrNestedReq.safeMinCountFulfilled / reqOrNestedReq.minCountRequired;
+              }
+            }
+          );
+        });
+        switch (r.groupName) {
+          case 'College':
+            this.hasCollege = true;
+            this.collegeRequirementCount.finished = totalCompleted;
+            this.collegeRequirementCount.needed = totalRequired;
+            break;
+          case 'Major':
+            this.hasMajor = true;
+            this.majorRequirementCount.finished = totalCompleted;
+            this.majorRequirementCount.needed = totalRequired;
+            break;
+          case 'Minor':
+            this.hasMinor = true;
+            this.minorRequirementCount.finished = totalCompleted;
+            this.minorRequirementCount.needed = totalRequired;
+            break;
+          default:
+            this.hasGrad = true;
+            this.gradRequirementCount.finished = totalCompleted;
+            this.gradRequirementCount.needed = totalRequired;
+            break;
+        }
+      });
+    },
+    sumRequirements(): number {
+      const totalRequired =
+        this.collegeRequirementCount.needed +
+        this.majorRequirementCount.needed +
+        this.minorRequirementCount.needed +
+        this.gradRequirementCount.needed;
+      const totalCompleted =
+        this.collegeRequirementCount.finished +
+        this.majorRequirementCount.finished +
+        this.minorRequirementCount.finished +
+        this.gradRequirementCount.finished;
+      return totalCompleted / totalRequired;
+    },
+  },
+  computed: {
+    progressValue(): string {
+      return 'transform: rotate(63deg)';
+      // return 45 + 1.8 * this['progress-info'];
+    },
+    requirements(): readonly GroupedRequirementFulfillmentReport[] {
+      return store.state.groupedRequirementFulfillmentReport;
+    },
+
+    getTransform(): string {
+      return 'transform: rotate(63deg)';
+    },
+    getImage(): string {
+      this.queryRequirements();
+      const progress = this.sumRequirements();
+      if (progress < 0.2) {
+        return this.progressImg.Hands;
+      }
+      if (progress < 0.4) {
+        return this.progressImg.Flex;
+      }
+      if (progress < 0.6) {
+        return this.progressImg.Star;
+      }
+      if (progress < 0.8) {
+        return this.progressImg.Fire;
+      }
```

Comment:
I think the following `if`s need to be `else if`s--it'll always return the fire emoji if `progress < 0.8`.